src/tester.rs

```diff
@@ -445,8 +446,8 @@ fn run_tests(
                     pass_stdout = Some(fuzzy::match_vec(stdout, &stdout_utf8));
                 }
 
-                // If something failed, we want to print out everything which either failed a test
-                // or wasn't part of a test.
+                // Second, if a test failed, we want to print out everything which didn't match
+                // (i.e. it either failed a test or wasn't part of a test).
```

Comment:
At the risk of sounding really anal, the wording "wasn't part of a test" is the part I found confusing, and it's cropped up again here in brackets. I'd prefer something like "it either caused the test to fail, or was not used in the matching process due to an earlier mismatch".

Reply:
How about:
"Second, if a test failed, we want to print out only the information releveant to the failure. For example, if the test doesn't specify matching rules for stderr, then don't print stderr".

Reply:
See if the new commit works for you.